execution/tests/log_capture.hpp

```diff
@@ -0,0 +1,35 @@
+#pragma once
```

Comment:
This utility is used in several tests that will be split between different modules in Leatherman. I wonder if there's a model we could use for sharing test utilities. @branan have any thoughts?